prov/verbs/src/ep_rdm/verbs_rdm.h

```diff
@@ -300,7 +301,7 @@ struct fi_ibv_rdm_ep {
 	int	n_buffs;
 	int	rq_wr_depth;    // RQ depth
 	int	sq_wr_depth;    // SQ depth
-	int	posted_sends;
+	ofi_atomic32_t	posted_sends;
 	int	posted_recvs;
```

Comment:
Does posted_recvs have the same issue?

Reply:
As you said, recv_preposted should be atomic as well, but this is not sufficient. For instance fi_ibv_rdm_batch_repost_receives() ignores bad_wr. recv_preposted should be fixed to the correct value when a partial wq list has been posted. So it seems there's also no point in pre-computing num_to_post.